core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -318,6 +318,8 @@ class ZkUtils(val zkClient: ZkClient,
 
     if (!zkClient.exists(path))
       ZkPath.createPersistent(zkClient, path, true, acl) //won't throw NoNodeException or NodeExistsException
+    else
+      ZkPath.setAcl(zkClient, path, acl)
```

Comment:
Do we need to do that? I thought this can be left with ZkSecurityMigrator. Otherwise, we will always have the overhead of setting the acl.